test/unittests/utils/test_utils_inputs.py

```diff
@@ -0,0 +1,591 @@
+
+    Notes
+    -----
+    This function is a simple prototype for creating fake datasets for testing.
+    It only adds (for now...) partial results of the t1-linear pipeline
+    for all subjects and sessions.
+    """
+    (directory / "subjects").mkdir()
+    for sub, sessions in subjects_sessions.items():
+        (directory / "subjects" / sub).mkdir()
+        for ses in sessions:
+            (directory / "subjects" / sub / ses).mkdir()
+            (directory / "subjects" / sub / ses / "t1_linear").mkdir()
+            (
+                directory
+                / "subjects"
+                / sub
+                / ses
+                / "t1_linear"
+                / f"{sub}_{ses}_T1w_space-MNI152NLin2009cSym_res-1x1x1_T1w.nii.gz"
+            ).mkdir()
+
+
+def rmtree(f: Path):
+    if f.is_file():
+        f.unlink()
+    else:
+        for child in f.iterdir():
+            rmtree(child)
+        f.rmdir()
+
+
+def test_insensitive_glob(tmp_path):
+    from clinica.utils.inputs import insensitive_glob
+
+    files = [
+        "foo.py",
+        "Bar.txt",
+        "bAZ.py",
+        "Fooo.PY",
+        "folder_1",
+        "folder_2",
+        "folder_1/foo1.py",
+        "folder_2/BaR2.PY",
+        "folder_1/baz.txt",
+    ]
+    for file in files:
+        d = tmp_path / file
+        d.mkdir()
+    python_files = insensitive_glob(str(tmp_path / "*.py"))
+    assert set([Path(f).name for f in python_files]) == {"foo.py", "bAZ.py", "Fooo.PY"}
+    text_files = insensitive_glob(str(tmp_path / "*.txt"))
+    assert set([Path(f).name for f in text_files]) == {"Bar.txt"}
+    assert len(insensitive_glob(str(tmp_path / "*.json"))) == 0
+    all_python_files = insensitive_glob(str(tmp_path / "**/*.py"), recursive=True)
+    assert set([Path(f).name for f in all_python_files]) == {
+        "foo.py",
+        "bAZ.py",
+        "Fooo.PY",
+        "foo1.py",
+        "BaR2.PY",
+    }
+
+
+def test_determine_caps_or_bids(tmp_path):
+    from clinica.utils.inputs import determine_caps_or_bids
+
+    assert not determine_caps_or_bids(tmp_path)
+    (tmp_path / "subjects").mkdir()
+    (tmp_path / "subjects" / "foo.txt").mkdir()
+    assert not determine_caps_or_bids(tmp_path)
+    (tmp_path / "subjects" / "sub-01").mkdir()
+    assert not determine_caps_or_bids(tmp_path)
+    (tmp_path / "groups").mkdir()
+    (tmp_path / "groups" / "foo.txt").mkdir()
+    assert not determine_caps_or_bids(tmp_path)
+    rmtree(tmp_path / "subjects")
+    (tmp_path / "sub-01").mkdir()
+    (tmp_path / "sub-01" / "foo.txt").mkdir()
+    assert determine_caps_or_bids(tmp_path)
+
+
+@pytest.mark.parametrize("folder_type", ["BIDS", "CAPS"])
+def test_common_checks(folder_type):
+    from clinica.utils.exceptions import ClinicaBIDSError, ClinicaCAPSError
+    from clinica.utils.inputs import _common_checks
+
+    with pytest.raises(
+        ValueError,
+        match="Argument you provided to ",
+    ):
+        _common_checks(1, folder_type)
```

Comment:
Expected type 'PathLike', got 'int' instead

Reply:
Looks like this is still unresolved

Reply:
Since the type mismatch is intentional:

```suggestion
        _common_checks(1, folder_type)  # noqa
```